File: src/AdventureWorks.Application/DataEngine/HumanResources/Shift/Commands/Queries/GetShifts/ShiftLookupModel.cs
using System;
using AutoMapper;
using AdventureWorks.Application.Interfaces.Mapping;
using AdventureWorks.BaseDomain.Entities.HumanResources;
using System.Collections.Generic;
using AdventureWorks.Common.Extensions;
using AdventureWorks.BaseDomain.Enums;
using AdventureWorks.BaseDomain.Interfaces;
using AdventureWorks.BaseDomain.CustomTypes;

namespace AdventureWorks.Application.DataEngine.HumanResources.Shift.Queries.GetShifts
{
    public partial class ShiftLookupModel : IHaveCustomMapping
    {
        public byte ShiftID { get; set; }
        public string Name { get; set; }
        public DateTime StartTime { get; set; }
        public DateTime EndTime { get; set; }
        public DateTime ModifiedDate { get; set; }

        #region Navigation Properties

        // Uncomment the following line if you need child navigation properties (can cause a poor performance)
        /*  */
        // Uncomment the following line if you need parent navigation properties
        /*  */
        #endregion

        public void CreateMappings(Profile configuration)
        {
            configuration.CreateMap<Domain.Entities.HumanResources.Shift, ShiftLookupModel>().IgnoreMissingDestinationMembers().CompensateWithDestinationValues();
            configuration.CreateMap<ShiftLookupModel, BaseShift>().IgnoreMissingDestinationMembers();
        }
    }
}

File: src/AdventureWorks.Web.Api/ClientApp/src/app/models/data/entities/human-resources/shift/shift-lookup-model.ts

export interface IShiftLookupModel {
  shiftID: number;
  name: string;
  startTime: Date | string;
  endTime: Date | string;
  modifiedDate: Date | string;

  // Uncomment the following line if you need child navigation properties (can cause a poor performance)
  /*  */
  // Uncomment the following line if you need parent navigation properties
  /*  */
}

export class ShiftLookupModel implements IShiftLookupModel {
  shiftID: number;
  name: string;
  startTime: Date | string;
  endTime: Date | string;
  modifiedDate: Date | string;

  // Uncomment the following line if you need child navigation properties (can cause a poor performance)
  /*  */
  // Uncomment the following line if you need parent navigation properties
  /*  */

  constructor(init?: Partial<ShiftLookupModel>) {
    if (!!init) {
      Object.assign<ShiftLookupModel, Partial<ShiftLookupModel>>(this, init);
    }
  }
}
